pkg/secrets/secrets.go

```diff
@@ -187,9 +183,63 @@ func SecretMounts(filePath, mountLabel, containerWorkingDir string) ([]rspec.Mou
 
 		mounts = append(mounts, m)
 	}
+
+	// Add FIPS mode secret
+	// The filePath check is to ensure that this secret is created only once
+	// as SecretMounts is called twice for the two mounts.conf files ("OverrideMountsFile" and "DefaultMountsFile") in the respective order
+	// therefore we want to add the FIPS mode secret, if it exists, after the secrets from the DefaultMountsFile are mounted
+	_, err := os.Stat("/etc/system-fips")
+	if err == nil && filePath == DefaultMountsFile {
```

Comment:
I think fips-mode should be set whether or not the User overrode the default mounts file.

Reply:
Right, it is always being set. This if code just makes sure this block of code is not called twice for no reason. The secrets code checks both the Override and Default Mounts files.